### PR TITLE
Add newer Elixir versions on CI

### PR DESCRIPTION
**Pull request by mopp**
## Overview

Please merge and review https://github.com/beam-telemetry/telemetry_metrics_statsd/pull/82 in advance.
I will rebase this PR after merging it.

I added newer Elixir versions on CI and fixed the warnings and errors.

## How to test

I confirmed that the CI was passed at the forked repository.
See https://github.com/mopp/telemetry_metrics_statsd/pull/3

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -42,7 +42,7 @@ defmodule TelemetryMetricsStatsd.MixProject do
       {:telemetry_metrics, "~> 0.6"},
       {:nimble_options, "~> 0.4 or ~> 1.0"},
       {:stream_data, "~> 0.4", only: :test},
-      {:dialyxir, "~> 0.5", only: :test, runtime: false},
+      {:dialyxir, "~> 1.3", only: :test, runtime: false},
```
Comment (mopp) [inline mix.exs:45]: I faced the error below on Elixir 1.15.x

```
> mix dialyzer
Checking PLT...
[:compiler, :elixir, :kernel, :liveness, :logger, :meck, :mock, :nimble_options,
 :stdlib, :stream_data, :telemetry, :telemetry_metrics, :tools]
Finding suitable PLTs
Looking up modules in dialyxir_erlang-26.0.1_elixir-1.15.0_deps-test.plt
Looking up modules in dialyxir_erlang-26.0.1_elixir-1.15.0.plt
Looking up modules in dialyxir_erlang-26.0.1.plt
Finding applications for dialyxir_erlang-26.0.1.plt
Finding modules for dialyxir_erlang-26.0.1.plt
Creating dialyxir_erlang-26.0.1.plt
:dialyzer.run error: Analysis failed with error:
{undef,[{prettypr,text,["'true'"],[]},
        {cerl_prettypr,lay_ann,2,[{file,"cerl_prettypr.erl"},{line,401}]},
        {cerl_prettypr,format,2,[{file,"cerl_prettypr.erl"},{line,261}]},
        {dialyzer_dataflow,signal_guard_failure,5,
                           [{file,"dialyzer_dataflow.erl"},{line,2477}]},
        {dialyzer_dataflow,bind_guard,3,
                           [{file,"dialyzer_dataflow.erl"},{line,1710}]},
        {dialyzer_dataflow,do_clause,7,
                           [{file,"dialyzer_dataflow.erl"},{line,1268}]},
        {dialyzer_dataflow,handle_clauses,9,
                           [{file,"dialyzer_dataflow.erl"},{line,1199}]},
        {dialyzer_dataflow,handle_case,3,
                           [{file,"dialyzer_dataflow.erl"},{line,932}]}]}
Last messages in the log cache:
  Reading files and computing callgraph... done in 0.16 secs
  Removing edges... done in 0.01 secs
Looking up modules in dialyxir_erlang-26.0.1.plt
** (Mix) Could not open /home/mopp/.mix/dialyxir_erlang-26.0.1.plt: no such file or directory

02:03:04.618 [error] Process #PID<0.1714.0> raised an exception
** (UndefinedFunctionError) function :prettypr.text/1 is undefined (module :prettypr is not available)
    :prettypr.text(~c"'true'")
    (dialyzer 5.1) cerl_prettypr.erl:401: :cerl_prettypr.lay_ann/2
    (dialyzer 5.1) cerl_prettypr.erl:261: :cerl_prettypr.format/2
    (dialyzer 5.1) dialyzer_dataflow.erl:2477: :dialyzer_dataflow.signal_guard_failure/5
    (dialyzer 5.1) dialyzer_dataflow.erl:1710: :dialyzer_dataflow.bind_guard/3
    (dialyzer 5.1) dialyzer_dataflow.erl:1268: :dialyzer_dataflow.do_clause/7
    (dialyzer 5.1) dialyzer_dataflow.erl:1199: :dialyzer_dataflow.handle_clauses/9
    (dialyzer 5.1) dialyzer_dataflow.erl:932: :dialyzer_dataflow.handle_case/3
```

https://github.com/jeremyjh/dialyxir/issues/478 might fix the error.
I just upgraded the dialyxir version.

**Comment by mopp** [comment]
@arkgil Please review and merge this PR also 🙏

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -41,6 +41,12 @@ jobs:
             elixir: 1.10.4
```
Comment (arkgil) [inline .github/workflows/test.yml:41]: I think we can remove testing on OTP 22.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -41,6 +41,12 @@ jobs:
...
           - otp: 24.0
             elixir: 1.12.0
+          - otp: 24.0
+            elixir: 1.13.0
+          - otp: 25.1
+            elixir: 1.14.0
+          - otp: 26.0
+            elixir: 1.15.4
```
Comment (arkgil) [inline .github/workflows/test.yml:49]:
```suggestion
          - otp: 24.3
            elixir: 1.13.4
          - otp: 25.3
            elixir: 1.14.5
          - otp: 26.0
            elixir: 1.15.4
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -79,6 +85,12 @@ jobs:
...
```
Comment (arkgil) [inline .github/workflows/test.yml:93]:
```suggestion
          - otp: 24.3
            elixir: 1.13.4
          - otp: 25.3
            elixir: 1.14.5
          - otp: 26.0
            elixir: 1.15.4
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (mopp) [reply .github/workflows/test.yml]: removed.
3756b85631b885dfcecbf8ba5fd270c29a8d635c

**Comment by ludwikbukowski** [comment]
nice!

**Pull request merged**